src/PrestaShopBundle/Service/Hook/HookDispatcher.php

```diff
@@ -35,6 +35,8 @@
  *
  * If the event is a RenderingHookEvent, then the final result is
  * an array of contents accessed from $event->getContent().
+ *
+ * @deprecated since 1.7.5, to be removed in 1.8. Use PrestaShop\PrestaShop\Core\Hook\HookDispatcher instead.
```

Comment:
If I read this correctly, you didn't deprecate it but only add an extra layer to the old implementation.

Deprecation means: "we don't use it anymore, but if external people use it, it will works as expected". This is not what is done here, because you re-use it.

Let's imagine we remove it: the new implementation will be broken because we depend on implementation and not to the contract.

What I suggest:

- Create `PrestaShop\PrestaShop\Adapter\Hook\HookDispatcher` with `PrestaShop\PrestaShopBundle\Service\Hook\HookDispatcher` content, this new class implements `HookDispatcherInterface`;
- Make `PrestaShop\PrestaShopBundle\Service\Hook\HookDispatcher` extends `PrestaShop\PrestaShop\Adapter\Hook\HookDispatcher`, empty it and add a constructor with the following content:

```php
public function __construct() {
    trigger_error('Deprecated since 1.7.5, to be removed in 1.8. Use PrestaShop\PrestaShop\Core\Hook\HookDispatcher instead.', E_USER_DEPRECATED);
parent::__construct();
}
- Now `PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface` can depend on his adapter (which is our strategy to deprecate legacy stuff) and in 1.8 we will be able to remove `PrestaShopBundle\Service\HookDispatcher` (not used anymore) for sure and maybe the adapter.
- you need to replace all calls of `PrestaShopBundle\Service\HookDispatcher` by the adapter.
```

WDYT?

Reply:
agree.